py/desispec/sky.py

```diff
@@ -982,25 +982,63 @@ def subtract_sky(frame, skymodel, apply_throughput_correction = False, zero_ivar
         log.error(message)
         raise ValueError(message)
 
-    if apply_throughput_correction and skymodel.throughput_corrections is not None :
-        # need to fit for a multiplicative factor of the sky model
-        # before subtraction
-        # we are going to use a set of bright sky lines,
-        # and fit a multiplicative factor + background around
-        # each of them individually, and then combine the results
-        # with outlier rejection in case a source emission line
-        # coincides with one of the sky lines.
-        for fiber in range(frame.flux.shape[0]) :
-            # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
-            skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
+    if skymodel.throughput_corrections is not None :
+        # a multiplicative factor + background around
+        # each of the bright sky lines has been fit.
+        # here we apply this correction to the emission lines only or to the whole
+        # sky spectrum
+        if apply_throughput_correction  :
+            for fiber in range(frame.flux.shape[0]) :
+                # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
+                skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
+
+        elif apply_throughput_correction_to_lines :
+
+            # we first identify bright lines on the sky model
+
+            # median sky of all fibers
+            msky  = np.median(skymodel.flux,axis=0)
+            # the continuum is the spectrum within 50 e/A of this interpolated minimal values
+            threshold = 50. # e/A
+
+            # find minima in sky spectrum
+            minima = np.zeros(msky.shape)
+            minima[1:-1] = (msky[1:-1]<msky[:-2])&(msky[1:-1]<msky[2:])&(msky[1:-1]>-20)&(msky[1:-1]<0.5*np.max(msky))
+            minima  = np.where(minima)[0]
+            if minima.size > 0 :
+                cont    = np.interp(skymodel.wave,skymodel.wave[minima],msky[minima])
+                # second minima
+                minima2 = np.zeros(msky.shape)
+                minima2[1:-1] = (cont[1:-1]<cont[:-2])&(cont[1:-1]<cont[2:])&(cont[1:-1]>-20)&(cont[1:-1]<0.5*np.max(cont))
+                minima2  = np.where(minima2)[0]
+                cont    = np.interp(skymodel.wave,skymodel.wave[minima2],msky[minima2])
+                in_cont = np.where(np.abs(msky-cont)<threshold)[0] # keep all pixels without n sigma of cont for debiasing estimate
+
```

Comment:
FWIW, I had trouble following the logic here.  Sometimes minimum_filter is nice for this kind of thing; I might have imagined defining minima as minima = msky == minimum_filter(msky, size=10) (a pixel is a 'minimum pixel' if it's the lowest within 10 pixels).  But what you're doing may well be better---I just am having problems thinking about it!

Reply:
this part of the code has been removed